apps/native/app/src/stores/auth-store.ts

```diff
@@ -207,23 +227,37 @@ export async function checkIsAuthenticated() {
         intl.formatMessage({ id: 'login.expiredTitle' }),
         intl.formatMessage({ id: 'login.expiredScopesMessage' }),
       )
+
       await logout()
+
       return false
     }
   }
 
-  fetchUserInfo().catch(async (err) => {
-    Alert.alert(
-      intl.formatMessage({ id: 'login.expiredTitle' }),
-      intl.formatMessage({ id: 'login.expiredMissingUserMessage' }),
-    )
-    await logout()
-    await Navigation.dismissAllModals()
-    await Navigation.dismissAllOverlays()
-    await Navigation.setRoot({
-      root: await getAppRoot(),
-    })
-  })
+  try {
+    await fetchUserInfo()
+
+    return true
+  } catch (e) {
+    const err = e as Error
 
-  return true
+    if (!offlineStore.getState().isConnected) {
+      return true
+    } else if (REFRESH_TOKEN_ERROR === err?.message) {
```

Comment:
~Instead of using offline store, which can have false positives or false negatives, I would prefer to actually handle different refresh responses propertly:

- actual "expired/invalid refresh token" errors should log the user out (401, invalid_grant, whatever).
- other refresh errors can be ignored (no internet, server down, 500, whatever).